display_list/display_list_image.h

```diff
@@ -54,6 +54,8 @@ class DlImage : public SkRefCnt {
   ///
   virtual std::shared_ptr<impeller::Texture> impeller_texture() const = 0;
 
+  virtual bool isOpaque() const = 0;
```

Comment:
We should document that this might conservatively return false when it cannot guarantee an opaque image. I'm guessing that it might return false for an ARGB image even if the image consists entirely of every pixel with an alpha value of 255, for instance. And some of our image types just can't tell. So we should include a doc comment here and mention the conservative intention of the return value.

Reply:
Skia doesn't treat this as whether the image contains any transparent pixels, but rather whether transparent pixels ignore the alpha because of the alpha type of the image info. I think documenting that this conservatively returns false is fine though.

Reply:
They have a really semantically odd way of combining pixel formats that may or may not refer to an alpha channel with an override setting saying whether the alpha is to be honored, but in the end it boils down to the combination of the two specifying a true pixel format and they don't necessarily honor your request to combine all SkColorType and SkAlphaType - they validate them, replacing some alpha types with others and just rejecting other combinations. In the end, if you created an info with (ARGB, Opaque_Alpha_Type) what you really did was to create an image that has no effective alpha channel and it's really an xRGB image with a really strange set of parameters to say that.

So, what I was saying is true - they are basing the opaque property on the type of the pixel format of the image (i.e. whether it has no alpha channel at all, or whether you specifically combined an alpha-bearing memory layout with a manual opaque override setting) - i.e. the overall combined pixel format is determining the answer.

And I did say that they don't care if an alpha image happens to have all opaque values, they only care what the format is. We're saying the same thing on that point.

We don't really expose pixel formats via DlImage, do we? If we do, I'd strongly recommend not following the confusing and somewhat self-contradicting Skia Color/AlphaType tuple.

Reply:
Added docs.